bot/cogs/custom_ui/page_turn_embed.py

```diff
@@ -0,0 +1,78 @@
+from typing import List
+
+import discord
+from discord.ext import commands
+
+
+class PageTurnView(discord.ui.View):
+    """discord.ui.View subclass for page-turn functionality between embeds
+
+    Parameters
+    ----------
+    ctx : discord.ext.commands.Context
+        Discord's command context
+
+    embed_list: List[discord.Embed]
+        List of embeds to page turn
+
+    message: discord.Message
+        The message containing the buttons.
+    timeout: float
+        (Optional) Float indicating seconds before timeout. (Default=60.0)
+    """
+
+    def __init__(
+        self,
+        ctx: commands.Context,
+        embed_list: List[discord.Embed],
+        message: discord.Message,
+        timeout=60.0,
+    ):
+        self.ctx = ctx
+        self.embed_list = embed_list
+        self.current_page = 0
+        self.message = message
+
+        super().__init__(timeout=timeout)
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        return interaction.user.id == self.ctx.message.author.id
+
+    # Remove buttons on timeout
+    async def on_timeout(self) -> None:
```

Comment:
There seems to be an issue here with timeouts. If you use the delete callback on the view before the timeout, the timeout will throw a `discord.errors.NotFound: 404 Not Found (error code: 10008): Unknown Message`. I'm unsure of the solution to this without investing more. I assume this is happening because the message was deleted but the timeout future task is still running.